service/src/main/java/com/codeforcommunity/requester/MapRequester.java

```diff
@@ -41,85 +46,132 @@ public MapRequester(Vertx vertx) {
         tokenFuture.compose(
             tokenString -> {
               MultiMap formData =
                   MultiMap.caseInsensitiveMultiMap()
                       .add("f", "json")
                       .add("token", tokenString)
-                      .add("updates", updateJson.encode());
+                      .add("where", "ID IN (" + String.join(",", blockIds) + ")")
+                      .add("outFields", "FID");
 
               return Future.future(
                   promise -> {
                     client
-                        .postAbs(featureLayerRoute)
+                        .postAbs(featureLayerQueryRoute)
                         .putHeader("content-type", "multipart/form-data")
                         .sendForm(
                             formData,
-                            (ar) -> {
-                              if (ar.succeeded()) {
-                                HttpResponse<Buffer> httpResponse = ar.result();
+                            (asyncResult -> {
+                              if (asyncResult.succeeded()) {
+                                HttpResponse<Buffer> httpResponse = asyncResult.result();
 
                                 if (httpResponse.statusCode() == 200) {
                                   JsonObject responseBody = httpResponse.bodyAsJsonObject();
-                                  if (responseBody.containsKey("error")
-                                      && responseBody.getJsonObject("error").getInteger("code")
-                                          == 498) {
-                                    // The API token is invalid, reset it and make this call again.
-                                    logger.info(
-                                        "Remaking token request while updating blocks to "
-                                            + updateTo.name());
-                                    this.tokenFuture = updateToken();
-                                    updateLayers(streetFids, updateTo, this.tokenFuture)
-                                        .onSuccess((V) -> promise.complete())
-                                        .onFailure(promise::fail);
-                                  } else if (responseBody.containsKey("updateResults")) {
-                                    logger.info(
-                                        "Successfully updated blocks to " + updateTo.name());
-                                    // Check successes
-                                    promise.complete();
-                                  } else {
-                                    // throw error
-                                    logger.error(
-                                        "Update street request responded with unrecognized response body: "
-                                            + responseBody);
+                                  if (responseBody.containsKey("features")) {
+                                    List<Map> features =
```

Comment:
Why are we using `List` and `Map`? Let's just use the JsonArray and JsonObject classes that the data is already in and that do the same thing so we don't do any raw typecasting